src/main.rs

```diff
@@ -106,3 +141,58 @@ fn print_category(
         }
     }
 }
+
+#[derive(Clone)]
+struct VersionTagParser;
+
+impl VersionTagParser {
+    fn make_error(cmd: &clap::Command, arg: Option<&clap::Arg>, value: &str) -> clap::Error {
+        let mut err = clap::Error::new(clap::error::ErrorKind::ValueValidation).with_cmd(cmd);
+        if let Some(arg) = arg {
+            err.insert(clap::error::ContextKind::InvalidArg, clap::error::ContextValue::String(arg.to_string()));
+        }
+        err.insert(clap::error::ContextKind::InvalidValue, clap::error::ContextValue::String(value.to_owned()));
+        err
+    }
+}
+
+impl clap::builder::TypedValueParser for VersionTagParser {
+
+    type Value = semver::Version;
+
+    fn parse_ref(
+        &self,
+        cmd: &clap::Command,
+        arg: Option<&clap::Arg>,
+        value: &std::ffi::OsStr,
+    ) -> Result<Self::Value, clap::Error> {
+
+        lazy_static! {
+            static ref RE: Regex = Regex::new(r"^v?(?P<version>\d+\.\d+\.\d+)$").unwrap();
+        }
+
+        let arg_value = value.to_string_lossy();
+
+        let result = RE
+            .captures(&arg_value)
```

Comment:
captures_iter maybe here to avoid the map? Alternatively wouldn't be a bad idea to maybe do:

```rs
if let RE.captures(&arg_value) = Some(captures) {
  //Do stuff here
}

avoids the flatten and I think it read a bit clearer than the nested map => extract => flatten setup and you save a bunch of addl maps 
```

Reply:
I’ll give it a shot.